### PR TITLE
frontend/pytorch: add aten::cross_entropy_loss translator and tests (fixes #29691, relates #28584)

### LINKED ISSUE #29691: [Good First Issue]: Support aten::cross_entropy_loss
### Context

As in the main issue.

### What needs to be done?

As in the main issue.

### Example Pull Requests

_No response_

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers
- [How to link your Pull Request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue-using-the-pull-request-sidebar)

### Contact points

As in the main issue.

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by Rahuldrabit**
### 1. Core Implementation
**File**: `src/frontends/pytorch/src/op/cross_entropy_loss.cpp`

Implements the `translate_cross_entropy_loss` function that converts PyTorch's cross_entropy_loss operation to OpenVINO operations. The implementation follows PyTorch's specification:

```
cross_entropy_loss = log_softmax + nll_loss
```

**Key features:**
- Applies `LogSoftmax` along dimension 1 (class dimension)
- Uses one-hot masking plus `ReduceSum` to select target class probabilities (NLL loss)
- Supports optional class weights via `Gather` + `Multiply`
- Supports three reduction modes:
  - `none` (0): No reduction, returns per-sample loss
  - `mean` (1): Mean of all losses (default)
  - `sum` (2): Sum of all losses
- Handles both 2D inputs `(N, C)` and higher-dimensional inputs `(N, C, d1, d2, ...)`
### 2. Operation Registration
**File**: `src/frontends/pytorch/src/op_table.cpp`

- Added forward declaration: `OP_CONVERTER(translate_cross_entropy_loss);`
- Registered operation: `{"aten::cross_entropy_loss", op::translate_cross_entropy_loss},`

### 3. Tests
**File**: `tests/layer_tests/pytorch_tests/test_cross_entropy_loss.py`

Comprehensive test suite covering:
- Basic functionality with different reduction modes (`mean`, `sum`, `none`)
- Class weights
- Different input shapes (2D and spatial dimensions)
- Edge cases (binary classification, single sample, many classes)

### CODE REVIEW

File: src/frontends/pytorch/src/op/cross_entropy_loss.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2018-2025 Intel Corporation
```
[the remark this answers is not in the document]
Comment (Rahuldrabit) [reply src/frontends/pytorch/src/op/cross_entropy_loss.cpp:1]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/openvinotoolkit/openvino/pull/32554#discussion_r2728794560)
